### PR TITLE
Add Internal Temperature usermod

### PR DESCRIPTION
**Pull request by dima-zhemkov**

### CODE REVIEW

**Comment by softhack007** [comment]
Is this the usermod from @lost-hope, or is it different?
https://github.com/MoonModules/WLED/pull/51

**Comment by dima-zhemkov** [comment]
> Is this the usermod from @lost-hope, or is it different? [MoonModules#51](https://github.com/MoonModules/WLED/pull/51)

It's based on corresponding usermod from MoonModules.

**Review by softhack007 — CHANGES_REQUESTED** [review]
Please clarify code ownership with @lost-hope, who created the same usermod in the MM fork.  I would prefer to have clear attribution of the code origin.

**Comment by lost-hope** [comment]
@blazoncek i PRed it into MM since it was requested by someone there. So why make a PR in AC, when it then has to be cherry picked into MM, when i could deliver it directly to MM for him. 

@dima-zhemkov it is ok here (not a lot of special and hard code in it), but you should ask the creator first instead of "getting caught.

@softhack007 see above ;D

**Comment by blazoncek** [comment]
> So why make a PR in AC, when it then has to be cherry picked into MM

Cherry picking only happens if there are conflicts or the maintainer doesn't want certain commits/code in their fork.

**Comment by dima-zhemkov** [comment]
@lost-hope I haven't "got caught" since I haven't broken any agreement. MIT license allows me to copy/reuse/modify any code from MoonModules/WLED repository without any additional permission of authors of that code as soon as a copy of the MIT license is in a destination repository. And I copied, literally, 5 lines of your code.
@softhack007 why did you ask me to clarify code ownership if you knew that it's under MIT license?

**Comment by dima-zhemkov** [comment]
So I should have been merely mention name of the author in a description?

File: usermods/Internal_Temperature_v2/usermod_internal_temperature.h
Comment (softhack007) [inline usermods/Internal_Temperature_v2/usermod_internal_temperature.h:9]: loopInterval should have a default value.
I would suggest something like 15 seconds.

File: usermods/Internal_Temperature_v2/usermod_internal_temperature.h
Comment (softhack007) [inline usermods/Internal_Temperature_v2/usermod_internal_temperature.h:11]: Should have a default value. Maybe start "disabled" by default, as it's possible that the undocumented temperature read function has unexpected side-effects.

File: usermods/Internal_Temperature_v2/readme.md
Comment (softhack007) [inline usermods/Internal_Temperature_v2/readme.md:11]: There should be some autor information here, plus some info about who is maintaining the usermod.

Please discuss with @lost-hope on what would be appropriate.

File: usermods/Internal_Temperature_v2/usermod_internal_temperature.h
Comment (softhack007) [inline usermods/Internal_Temperature_v2/usermod_internal_temperature.h:64]: Adding to the info page only makes sense if the usermod is enabled.
You could add `if (! isEnabled) return;`

File: usermods/Internal_Temperature_v2/usermod_internal_temperature.h
Comment (softhack007) [inline usermods/Internal_Temperature_v2/usermod_internal_temperature.h:80]: @blazoncek : better to have this methods as an "empty body", or better to delete empty methods, so the usermod will not overwrite the "upper class" default methods?

**Review by softhack007 — CHANGES_REQUESTED** [review]
See detailed code comments.

File: usermods/Internal_Temperature_v2/usermod_internal_temperature.h
Comment (softhack007) [inline usermods/Internal_Temperature_v2/usermod_internal_temperature.h:44]: How accurate is the temperature function? Is it justified to have two decimal places, i.e. is the temperature value accurate up to 1/100th?

File: usermods/Internal_Temperature_v2/usermod_internal_temperature.h
Comment (softhack007) [inline usermods/Internal_Temperature_v2/usermod_internal_temperature.h:77]: Any plans to also support °F (Fahrenheit)?

Comment (dima-zhemkov) [reply usermods/Internal_Temperature_v2/usermod_internal_temperature.h:77]: Nope

File: usermods/Internal_Temperature_v2/usermod_internal_temperature.h
Comment (softhack007) [inline usermods/Internal_Temperature_v2/usermod_internal_temperature.h:65]: Why `lightArr` ? Can you also measure light here?

File: usermods/Internal_Temperature_v2/usermod_internal_temperature.h
Comment (blazoncek) [reply usermods/Internal_Temperature_v2/usermod_internal_temperature.h:80]: Better to remove these and rely on virtual ones from base class.
If these were required they would be pure virtual in base class.

File: usermods/Internal_Temperature_v2/usermod_internal_temperature.h
Comment (dima-zhemkov) [reply usermods/Internal_Temperature_v2/usermod_internal_temperature.h:44]: Not so accurate actually. I think one decimal place will be enough.

File: wled00/usermods_list.cpp
Comment (softhack007) [inline wled00/usermods_list.cpp:177]: Please agree with @lost-hope on a folder and name for this usermod (in MM fork its usermods/mcu_temp). It might be a minor point for this PR to "Aircoookie" WLED, but we should do our best to avoid having two usermods with almost identical code but different name and folder.

File: usermods/Internal_Temperature_v2/usermod_internal_temperature.h
Comment (dima-zhemkov) [reply usermods/Internal_Temperature_v2/usermod_internal_temperature.h:9]: It has a default value, see line 99.

File: usermods/Internal_Temperature_v2/usermod_internal_temperature.h
Comment (dima-zhemkov) [reply usermods/Internal_Temperature_v2/usermod_internal_temperature.h:11]: It has a default value, see line 98.

File: usermods/Internal_Temperature_v2/usermod_internal_temperature.h
Comment (softhack007) [inline usermods/Internal_Temperature_v2/usermod_internal_temperature.h:98]: @blazoncek is it sufficient to have a default value here, but omit a default when declaring the variable on line 9 and line 10?

File: usermods/Internal_Temperature_v2/usermod_internal_temperature.h
Comment (softhack007) [inline usermods/Internal_Temperature_v2/usermod_internal_temperature.h:99]: @lost-hope if I remember correctly, there were some instabilities when reading the sensor too often.... what would you suggest as a default frequency for polling the temperature? 15 second is maybe?

File: usermods/Internal_Temperature_v2/usermod_internal_temperature.h
Comment (blazoncek) [reply usermods/Internal_Temperature_v2/usermod_internal_temperature.h:98]: I've seen some usermods do that. I prefer constructor and have updated a few usermods accordingly.

File: usermods/Internal_Temperature_v2/usermod_internal_temperature.h
Comment (dima-zhemkov) [reply usermods/Internal_Temperature_v2/usermod_internal_temperature.h:63]: Done

File: usermods/Internal_Temperature_v2/usermod_internal_temperature.h
Comment (dima-zhemkov) [reply usermods/Internal_Temperature_v2/usermod_internal_temperature.h]: Done

File: usermods/Internal_Temperature_v2/usermod_internal_temperature.h
Comment (dima-zhemkov) [reply usermods/Internal_Temperature_v2/usermod_internal_temperature.h:40]: Changed to one decimal place

File: usermods/Internal_Temperature_v2/usermod_internal_temperature.h
Comment (dima-zhemkov) [reply usermods/Internal_Temperature_v2/usermod_internal_temperature.h]: Fixed

File: usermods/Internal_Temperature_v2/usermod_internal_temperature.h
Comment (dima-zhemkov) [reply usermods/Internal_Temperature_v2/usermod_internal_temperature.h:89]: I can replace default values with field initializers, but I don't see any reason to create a constructor

File: wled00/usermods_list.cpp
Comment (dima-zhemkov) [reply wled00/usermods_list.cpp:177]: I don't want to waste @lost-hope's time. After we merge this PR I will create new one in the MM fork and will replace usermods/mcu_temp with this usermod.

File: usermods/Internal_Temperature_v2/readme.md
Comment (dima-zhemkov) [reply usermods/Internal_Temperature_v2/readme.md:11]: @softhack007 perhaps you have a template or an example of how it should look like?

@lost-hope any thoughts about this?

File: usermods/Internal_Temperature_v2/usermod_internal_temperature.h
Comment (dima-zhemkov) [reply usermods/Internal_Temperature_v2/usermod_internal_temperature.h]: I've changed default interval to 10 seconds

File: usermods/Internal_Temperature_v2/usermod_internal_temperature.h
Comment (dima-zhemkov) [reply usermods/Internal_Temperature_v2/usermod_internal_temperature.h]: Done

File: usermods/Internal_Temperature_v2/readme.md
Comment (blazoncek) [reply usermods/Internal_Temperature_v2/readme.md:11]: If in doubt put both authors in. Possibly with email or Github nickname & profile link.

Comment (dima-zhemkov) [reply usermods/Internal_Temperature_v2/readme.md]: Done

**Comment by blazoncek** [comment]
As far as I am concerned this is ok.
@lost-hope do you agree with this PR? I see you are credited.
@softhack007 do you have any more objections?

**Comment by lost-hope** [comment]
Fine with me

**Review by softhack007 — APPROVED** [review]
Somehow my last review was lost after force-push 🙄 

> Sometimes commits can disappear after a force-push.

Ok for me with the latest changes.

**Pull request merged**
